File: repo/atlas-foundry/examples/create-token.java
package com.atlas402.foundry.examples;

import com.atlas402.foundry.core.*;

public class CreateToken {
    public static void main(String[] args) {
        AtlasFoundry foundry = new AtlasFoundry("https://facilitator.payai.network");
        
        TokenParams params = new TokenParams(
            "My Service Token",
            "MST",
            "1000000",
            "10.00",
            "base",
            "0x1234567890123456789012345678901234567890"
        );
        
        TokenResult token = foundry.createToken(params).join();
        System.out.println("Token created: " + token.getContractAddress());
    }
}

File: repo/atlas-foundry/examples/create-token.ts
import { AtlasFoundry } from '../core/foundry';
import { createWalletClient, http } from 'viem';
import { base } from 'viem/chains';

async function main() {
  const walletClient = createWalletClient({
    chain: base,
    transport: http(),
  });

  const foundry = new AtlasFoundry({
    facilitatorUrl: 'https://facilitator.payai.network',
  });

  await foundry.initializeEVM(walletClient);

  const token = await foundry.createToken({
    name: 'My Service Token',
    symbol: 'MST',
    supply: '1000000',
    pricePerMint: '10.00',
    network: 'base',
    deployerAddress: '0x1234567890123456789012345678901234567890',
  });

  console.log('Token created:', token);
}

main().catch(console.error);

